### DISCUSSION TITLE
Aktualny plan studiów vs. nowy plan studiów - Semestr 1

### DISCUSSION THREAD

**Discussion by PiotrAb** [IS-UMK/sylabusy#2] [Ideas]
Michałowie,

wiem, że odciąganie od rewolucji nie jest w moim stylu (zawsze byłem tym podżegającym do usiłowania), ale chciałbym abyśmy wypracowali pewien satysfakcjonujący konsensus, który nie będzie wymagał od nas odrzucania tego co obecnie jest wypracowane i w sumie wiemy, że mamy kim obsadzić zajęcia.

Spójrzmy na [Plan Studiów 2024/25](https://www.fizyka.umk.pl/panel/wp-content/uploads/InformatykaStosowana_s1_2024_25.pdf) :

## Semestr 1:
 - BHP (8h) - nie do ruszenia
 - WDoS (16h) - nie do ruszenia
 - Analiza 1 ( 20w + 50ć = 80h) - cała masa godzin, na ćwiczeniach można dostosować SYLABUS do naszyh potrzeb, wykład to 20h w trakcie których da się zrobić jedynie podstawy
 - Algebra 1 ( 20w + 40ć = 60h) - bardzo podobna sytuacja, ćwiczenia mocno ratują sprawę i pozwalają na dostosowanie treści pod nasze potrzeby
 - Wstęp do UNIXa (15w + 30lab = 45h) - chyba nie trzeba nic komentować, jest OK
 - Programowanie Proceduralne ( 30w + 30lab = 60h) - tutaj sporo treści pokrywa się z Językami Programowana na sem. 2 i moim zdaniem można by było więcej czasu poświęcić tutaj na to, co według Waszych notatek znajduje się we `Wstępie do informatyki`
 - Fizyka dla informatyków cz. 1 ( 30w + 30ćw. = 60h) - z jednej strony trochę zapychacz, z drugiej moim zdaniem istotny przedmiot dla przyszłego inżyniera pod warunkiem, że będzie właściwie prowadzony - treści mogą być sensowne
 - OPSS cz. 1 - (15w + 20 lab = 35h) - wstęp do podstaw budowy UI - HTML + CSS + biblioteki CSS 

Sensownych zajęć jest 340h, 24h to konieczne zapychacze. 
ok. 23h tygodniowo, niecałe 5h na dzień - semestr 1 nie jest tragicznie zapchany.

Moje uwagi: po przejrzeniu sylabusów - duża część `programowania proceduralnego` pokrywa się z późniejszymi `językami programowania` - wydaje mi się, że tutaj mamy pole do popisu. Drugi przedmiot to ta nieszczęsna `fizyka dla informatyków cz. 1` - patrząc po sylabusie to taka trochę bajeczka o wszystkim i o niczym, a więc może warto by było to nakierować na jakieś bardziej przydatne zagadnienia (np. proste całkowanie równań ruchu, wykorzystanie jakiegoś prostego silnika do implementacji fizyki itp.)

**Comment by PiotrAb** [comment]
Chciałbym, aby moje podsumowanie było próbą podjęcia kroków zmierzających do reformy, powolnej ale sensownej. Wydaje mi się, że można zrobić coś z sensem bez konieczności przewalania całej pracy osób, które już te siatki przygotowały.

Zobaczcie - na przykład zmiana `Programowania Proceduralnego` na `Wstęp do Informatyki` i drobna korekcja treści bardzo mocno zmienia cały punkt widzenia - mamy miejsce na treści, których nam brakuje (więcej git'a, GH i narzędzi programistycznych).

**Comment by mgrochowski** [comment]
> Moje uwagi: po przejrzeniu sylabusów - duża część `programowania proceduralnego` pokrywa się z późniejszymi `językami programowania` - wydaje mi się, że tutaj mamy pole do popisu. 

Warto pamiętać, że Programowanie Proceduralne i Języki Programowania należy traktować jako jeden przedmiot podzielony na 2 części. Jeśli niektóre treści się pokrywają to zapewne dlatego, że na Programowaniu Proceduralnym te zagadnienia są CELOWO przestawione w bardzo okrojonej formie (w formie wprowadzenia) i dopiero na Językach Programowania są uzupełnione i rozszerzone. Taki był zamysł od początku i program obu przedmiotów był uzgadniany.

**Reply by PiotrAb** [reply]
Jasne, rozumiem, bo nawet wychodzi z tego jedna ze ścieżek zwiazanych z programowaniem i w tym przypadku spiralny program nauczania ma jak najbardziej sens. 

Uważam natomiast, że kosztem np. wskaźników w PP, które później tłuczone są na JPr oraz na przykład zagadnień z reprezenatcją symboli, które powtarzają się na Technice Komputerowej w semestrze 2, można by było rozszerzyć sprawy związane z GitHub'em, dobrymi praktykami i np. CI/CD. Tak, wiem, że później jest kurs Narzędzia Programistyczne, ale jest to kurs do wyboru, na który nie każdy ze Studentów dociera...

Zobacz Marku - podobną sytuację mamy z SO - lwia część początkowych zagadnień poruszana jest na Technice komputerowej i w sumie planowaliśmy zapoznanie się z materiałem prof. Strzałkowskiego, żeby uniknąć tak licznych powtórek i chcieliśmy trochę czasu na tym zaoszczędzić i przesunąć w bardziej istotne zagadnienia związane z SO.

Moja sugestia wynika z tego, że dość mocno powstrzymywałbym wszystkich przez rewolucją w programach studiów, bo wiem, że napotkamy opory, a techniką małych kroków możemy na prawdę w ciekawy sposób zmodernizować aktualne plany studiów.
